### PR TITLE
fixed exec issue for restricted systems

### PR DESCRIPTION
**Pull request by AlexeyALeonov**
Related to #23

How to test:
* build the docker image:

```
docker buildx build -f Dockerfile -t node:latest .
```
* Create an image file and mount it with `noexec` to emulate a restricted mount:

```
truncate -s 100M temp.img
mkfs -t ext4 temp.img
mkdir temp-disk
sudo mount -o rw,noexec temp.img temp-disk
sudo chown $(id -i):$(id -g) temp-disk
```

## testing a normal flow
1. Run the container, and wait till the binaries are downloaded:
```
docker run --rm --name test --mount type=bind,source="${PWD}/test-disk",destination=/app/config node:latest
```
2. Check that there were no permission errors
3. Check that the `test-disk` contains both binaries in "test-disk/bin"
4. Rerun the container; binaries should not be downloaded again, and the container should be running without permission errors.

## storagenode has been updated in the ephemeral storage, but not in a persistent storage
1. download an old storagenode binary
```
wget -O /tmp/storagenode.zip https://github.com/storj/storj/releases/download/v1.111.4/storagenode_linux_amd64.zip
mkdir -p test-disk/bin
unzip -p /tmp/storagenode.zip > test-disk/bin/storagenode
chmod u+x test-disk/bin/storagenode
```
2. run the container
```
docker run --rm --name test --mount type=bind,source="${PWD}/test-disk",destination=/app/config node:latest
```
3. check that a new version of storagenode has been downloaded and replaced both in the persistent and ephemeral storages
4. check that there were no permissions errors
5. Check that the `test-disk` contains both binaries in "test-disk/bin"
6. Rerun the container; binaries should not be downloaded again, and the container should be running without permission errors.
7. Make sure, that the old version is updated to a minimum version, if the update to the suggested version is not allowed yet. 

## cleanup

```
sudo umount test-disk
rm -rf test-disk test.img
```

### CODE REVIEW

**Review by profclems — COMMENTED** [review]
This is a good change in the right direction but the problem is that the updater is only aware of the BINARY_DIR location and won't copy back the updated binary into the `BINARY_STORE`. On restart, the old binaries will be run

**Comment by AlexeyALeonov** [comment]
Please, provide your suggestions.
Right now it would behave like this:
If the new version is arriving while the node is running, it will update only the internal path `${BINARY_DIR}`. 
If the old version would fail with an exception, the container will be restarted and heal itself. If there's no exception, it will be updated as usual.
Only the restart of the container will update the `${BINARY_STORE}` in this case.

**Comment by AlexeyALeonov** [comment]
Please check again, all edge cases are covered.

File: Dockerfile [hunk 1]
```diff
@@ -37,4 +37,5 @@ ENV ADDRESS="" \
     SETUP="false" \
     AUTO_UPDATE="true" \
     LOG_LEVEL="" \
-    BINARY_DIR="/app/config/bin"
+    BINARY_DIR="/app/bin" \
+    BINARY_STORE="/app/config/bin"
```
Comment (profclems) [inline Dockerfile]: Let's make it `BINARY_STORE_DIR` so it will reflect the new flag `--binary-store-dir` of the updater later.

File: Dockerfile [hunk 2]
```diff
@@ -37,4 +37,5 @@ ENV ADDRESS="" \
     SETUP="false" \
     AUTO_UPDATE="true" \
     LOG_LEVEL="" \
-    BINARY_DIR="/app/config/bin"
+    BINARY_DIR="/app/bin" \
```
Comment (profclems) [inline Dockerfile]: nit: I'm thinking we don't even need this env anymore since we are somehow going back to the default. We only need the BINARY_STORE. In that case we wouldn't have to worry so much about changing the /app/bin in every file. Also to avoid confusing users about which one is which

**Review by profclems — DISMISSED** [review]
This looks good except with a few nit comments from me

**Review by profclems — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (profclems) [reply Dockerfile]: Added a commit to fix that

File: Dockerfile (hunk 2 above)
Comment (profclems) [reply Dockerfile]: Added a commit to fix that

**Comment by profclems** [comment]
I did a couple of tests locally with all edge cases I could think of and it looks good. Thanks @AlexeyALeonov
